src/backend/distributed/commands/multi_copy.c

```diff
@@ -3497,22 +3564,17 @@ InitializeCopyShardState(CopyShardState *shardState,
 		{
 			shardState->copyOutState = (CopyOutState) palloc0(sizeof(*copyOutState));
 			CloneCopyOutStateForLocalCopy(copyOutState, shardState->copyOutState);
-			LogLocalCopyExecution(shardId);
-			continue;
-		}
 
-
-		if (placement->groupId == GetLocalGroupId())
-		{
-			/*
-			 * if we are copying into an intermediate file we won't use local copy.
-			 * Files are visible to all transactions so we can still use local execution, therefore
-			 * we shouldn't restrict only using connection in this case.
-			 */
-			if (!isCopyToIntermediateFile)
+			if (colocatedIntermediateResult)
 			{
-				SetLocalExecutionStatus(LOCAL_EXECUTION_DISABLED);
+				LogLocalCopyToFileExecution(shardId);
 			}
+			else
+			{
+				LogLocalCopyExecution(shardId);
```

Comment:
This method sounds like it is a general one and there is also `LogLocalCopyToFileExecution`, would that make sense to have a single method and pass `shardId` and `copyToFile` flag and then generate the message based on these two parameters. That way the message would also be consistent and we would have less code.

Reply:
I also don't like passing boolean flags, yet I still don't know how to not copy paste code without doing that :)

Lets keep them the way they are but I guess in that case we should rename LogLocalCopyExecution, because it currently could also mean copy data to a local file?